semantic_model_generator/validate/context_length.py

```diff
@@ -1,13 +1,66 @@
-from typing import TypeVar
+from typing import Any, TypeVar
 
 from google.protobuf.message import Message
 from loguru import logger
 
 from semantic_model_generator.data_processing.proto_utils import proto_to_yaml
 
 ProtoMsg = TypeVar("ProtoMsg", bound=Message)
-_MODEL_CONTEXT_LENGTH_TOKENS = 6500  # We use 6.5k, with 1.2k for instructions, so that we can reserve 500 for response tokens (average is 300).
-_MODEL_CONTEXT_INSTR_TOKEN = 20  # buffer for instr tokens
+
+# Max total tokens is 8200.
+# We reserve 500 tokens for response (average response is 300 tokens).
+# So the prompt token limit is 7700.
+_TOTAL_PROMPT_TOKEN_LIMIT = 7700
+_BASE_INSTRUCTION_TOKEN_LENGTH = 1220
+_TOKENS_PER_LITERAL_RETRIEVAL = 100
```

Comment:
is this the token per literal/sample value? feels quite high?

Reply:
right now we're retrieving 10 literals per cortex search service call. I estimated 10 tokens per retrieved literal, since each literal is presented as a filter expression (i.e. `table.column = 'literal'`). So 10 literals * 10 tokens per literal gives 100 tokens per literal retrieval call.

Reply:
got it! Let's update 10 to be a constant with comment on the assumption

Reply:
done